CHARTER.md

```diff
@@ -0,0 +1,117 @@
+
+* The Crossplane Resource Model (XRM). An API model for Crossplane extensions.
+* The ability to compose bespoke control plane APIs for any compute plane.
+* A package format and manager to extend your control plane with:
+  * The ability to orchestrate any kind of compute plane provider.
+  * New ways to compose control plane APIs.
+  * New integration with supporting services such as secret stores.
+* Tooling to create and test new Crossplane extensions.
+* A conformance program for Crossplane extensions.
+
+As a framework, Crossplane:
+
+* Avoids building features specific to any particular compute plane in its core.
+  Instead features pertaining to any particular cloud provider, project, or tool
+  are enabled through extension points.
+* Focuses on APIs. Crossplane provides and recommends “golden path” tools to
+  build conformant extensions but does not require them. Anything with
+  conformant API schema and behavior may be considered a Crossplane extension -
+  regardless of how it is built. 
+* Optimizes for “vertical integration”. That is, Crossplane extensions are
+  intended and designed to be used in conjunction with Crossplane and each
+  other.
+* Prioritizes enabling a “separation of concerns” between control plane curators
+  and consumers. Curators deploy, extend, and configure control planes, while
+  consumers use them to orchestrate compute planes.
+
+## What Crossplane Is Not
+
+Crossplane is not intended to be a “batteries included” control plane for any
+particular use case. Instead it is a framework that you can deploy, configure,
+and extend to power control planes for your own highly bespoke use cases, often
+without writing any code.
+
+The project is a neutral place for vendors and individuals to come together in
+enabling control planes. To that extent, it does not police Crossplane
+extensions. Any extension that passes the appropriate Crossplane conformance
+suite may consider itself to be “a Crossplane extension” - regardless of how it
+is built, how it is licensed, who maintains it, how it is governed, or where it
+is hosted. Conformance suites are typically concerned with behavior and (where
+applicable) API shape, not internal implementation details.
+
+Concrete examples of things that are not in scope for the project include (but
+are not limited to):
+
+* Providing or mandating abstractions around specific use cases such as
+  provisioning applications (e.g. a PaaS API) or provisioning Kubernetes
+  clusters. 
+* Providing or mandating a configuration language (for example, CUE). Crossplane
+  provides a declarative API that may be targeted by arbitrary configuration
+  languages, and extension points that may be powered by arbitrary languages.
+* Enabling the use of Crossplane extensions without using Crossplane and vice
+  versa; this is not supported and may result in undefined behavior.
+* Providing or mandating opinions around control plane hosting and tenancy
+  models.
```

Comment:
Thinking about this in the context of our current "multi-tenancy" features, do this mean something like provider partitioning in core Crossplane would be out of scope? It is related to multi-tenancy, and saying multi-tenancy features will not be "provided" feels like a pretty strong statement around what we won't do. Perhaps what I'm looking for is whether "interfaces the enable multi-tenancy" may be in scope, rather than concrete implementations. An interface could be considered an "opinion" about multi-tenancy.

Reply:
I don't love this bullet point as it stands - I've struggled to succinctly capture our thinking around tenancy. This is complicated because "tenants" can mean different things to different people. Different tenants could be different apps, different teams, or different paying customers.

Today I believe Crossplane (like Kubernetes) offers a kind of "soft" or "leaky" multi-tenancy. Claims are namespaced, but anyone who wants to discover available Compositions or "see how the sausage is made" (i.e. how their claim is composed) needs at least read-only RBAC access to Compositions and/or some kinds of MR at the cluster scope. I'd judge this to be fairly tolerable for _team based_ tenancy - it's probably not so bad that you can see what infrastructure another team has - but not acceptable for paying customers. I'd say Kubernetes is pretty similar - you'd probably let many teams have `kubectl` access to a single cluster but not multiple paying customers.

As you allude to, Crossplane has further tenancy constraints in the sense that some orgs want to use pod based identity (e.g. AWS IRSA) _and_ multiple AWS accounts for multiple tenants.

With all that context in place, the spirit of what I'm trying to say here is that enabling a stronger tenancy model is out of scope - I think that would include adding interfaces to enable that model. We feel that (again similar to Kubernetes) the Crossplane tenancy model is good enough for many use cases, and that where it isn't good enough the best approach is to deploy multiple Crossplanes.